de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/controller/ProjectExplorerViewController.java

```diff
@@ -118,29 +129,92 @@ public void initializeView(IEclipseContext context, WorkspaceFileSystem fileSyst
 			}
 		});
 
+		// Handoff Toolbar
+		toolbarController = new ProjectExplorerToolBarController(projectToolbar, services,
+				canvas.getParent().getShell());

```

Comment:
We can open the context menu with right clicking on the toolbar (because we register it on the canvas here).
Maybe we can only use it on the tree with adding a subcanvas or something like that.

Reply:
I tested that and attaching the context menu directly onto the project tree object but without success.
The behavior is weird but not application breaking so I propose ignoring it.

Reply:
Yeah I thought about it again. I think the only way is we possible could resolve this is using two SWT canvas for each the toolbar and the tree that would have to be split in different fxml files, which kinda sucks. So lets leave it for now...